### PR TITLE
AUD based indices

### LINKED ISSUE #1103: Add AUD indices
AONIA, BBSW 3M and 6M

### PR DESCRIPTION
**Pull request by yukiiwashita**
* Added Overnight index and Ibor indices.
* Added Sydney calendar Id, but the default in GlobalHolidayCalendars is weekend only holiday calendar, needs to be implemented. 
* This work closes #1103

### CODE REVIEW

File: modules/basics/src/main/java/com/opengamma/strata/basics/index/OvernightIndices.java [hunk 1]
```diff
@@ -51,6 +51,12 @@
    * FED_FUND is an "Overnight" index.
    */
   public static final OvernightIndex USD_FED_FUND = OvernightIndex.of("USD-FED-FUND");
+  /**
+   * The AUD_RBAON index for AUD.
+   * <p>
+   * AUD_RBAON is an "Overnight" index.
+   */
+  public static final OvernightIndex AUD_RBAON = OvernightIndex.of("AUD-RBAON");
```
Comment (jodastephen) [inline modules/basics/src/main/java/com/opengamma/strata/basics/index/OvernightIndices.java]: ISDA uses "AONIA"

File: modules/basics/src/main/java/com/opengamma/strata/basics/date/GlobalHolidayCalendars.java [hunk 2]
```diff
@@ -682,6 +691,11 @@ static ImmutableHolidayCalendar generateOslo() {
     return ImmutableHolidayCalendar.of(HolidayCalendarId.of("NOOS"), holidays, SATURDAY, SUNDAY);
   }
 
+  static ImmutableHolidayCalendar generateSydney() {
+    List<LocalDate> holidays = new ArrayList<>(0); //TODO standard calendar should be implemented
```
Comment (jodastephen) [inline modules/basics/src/main/java/com/opengamma/strata/basics/date/GlobalHolidayCalendars.java]: Not sure this should be merged without a holiday calendar

**Comment by jodastephen** [comment]
Names also need to be added to `FloatingRateNameData.ini` from [FpML](http://www.fpml.org/coding-scheme/floating-rate-index-2-14.xml).

File: modules/basics/src/main/resources/com/opengamma/strata/config/base/FloatingRateNameData.ini [hunk 3]
```diff
@@ -89,6 +89,8 @@ USD-FEDFUND-H.15-OIS-COMPOUND = USD-FED-FUND
 CAD-CORRA = CAD-CORRA
 CAD-CORRA-OIS-COMPOUND = CAD-CORRA
 
+AUD-AONIA = AUD-AONIA
```
Comment (jodastephen) [inline modules/basics/src/main/resources/com/opengamma/strata/config/base/FloatingRateNameData.ini]: Need to add to two sections. Under the [ibor] section:

```
AUD-BBSW = AUD-BBSW
AUD-BBR-AUBBSW = AUD-BBSW
AUD-BBR-BBSW = AUD-BBSW
AUD-BBR-BBSW-Bloomberg = AUD-BBSW
AUD-BBR-BBSW-Bloomberg = AUD-BBSW
```

and under [overnightCompounded] section:

```
AUD-AONIA = AUD-AONIA
AUD-AONIA-OIS-COMPOUND = AUD-AONIA
```

File: modules/basics/src/main/resources/com/opengamma/strata/config/base/IborIndexData.csv [hunk 4]
```diff
@@ -109,3 +109,7 @@ NOK-NIBOR-1M,NOK,true,Act/360,NOOS,2,NOOS,NOOS,1M,LastBusinessDay,12:00,Europe/O
 NOK-NIBOR-2M,NOK,true,Act/360,NOOS,2,NOOS,NOOS,2M,LastBusinessDay,12:00,Europe/Oslo
 NOK-NIBOR-3M,NOK,true,Act/360,NOOS,2,NOOS,NOOS,3M,LastBusinessDay,12:00,Europe/Oslo
 NOK-NIBOR-6M,NOK,true,Act/360,NOOS,2,NOOS,NOOS,6M,LastBusinessDay,12:00,Europe/Oslo
+,,,,,,
+AUD-BBSW-3M,AUD,true,Act/365F,AUSY,1,AUSY,AUSY,3M,LastBusinessDay,11:00,Australia/Sydney
```
Comment (marc-opengamma) [inline modules/basics/src/main/resources/com/opengamma/strata/config/base/IborIndexData.csv]: BBSW exists for 1m to 6m (see http://www.afma.com.au/data/bbsw). We should have all of them in the list.

**Pull request merged**